src/vis/vis.js

```diff
@@ -471,57 +461,42 @@ var Vis = cdb.core.View.extend({
       this.mapView.bind('newLayerView', this.addTooltip, this);
     }
 
-    // Create an instance of a map (datasource)
-    var windshaftMap;
-    if (data.datasources && data.datasources.length) {
-      // Only one datasource?
-      var datasource = _.first(data.datasources);
-      var datasourceLayer = _.find(data.layers, function(l) {
-        return l.datasource === datasource.id
-      });
+    this.map.layers.reset(_.map(data.layers, function(layerData) {
+      var model;
 
-      var layerGroupLayer = data.layers[1];
-      var windshaftClient = new cdb.windshaft.Client({
-        ajax: $.ajax,
-        user_name: datasource.user_name,
-        maps_api_template: datasource.maps_api_template,
-        stat_tag: datasource.stat_tag,
-        force_compress: false,
-        force_cors: false,
-        endpoint: MapBase.BASE_URL // This is different for named_maps
-      });
+      // TODO: 'named_map' ????
+      if (layerData.type === 'layergroup') {
 
-      var layerDefinition = new LayerDefinition(datasourceLayer.options.layer_definition, {}, data.widgets);
-      windshaftMap = windshaftClient.instantiateMap(layerDefinition);
+        var cartoDBLayers = _.map(layerData.options.layer_definition.layers, function(layer) {
+          return Layers.create(layer.type, self, layer);
+        });
 
-      this.datasource = new cdb.core.Datasource(datasource, { windshaftMap: windshaftMap });
+        model = new cdb.geo.CartoDBGroupLayer({}, {
+          layers: cartoDBLayers
+        });
 
-      _.each(data.widgets, function(d) {
-        var opts = d.options;
-        var type = d.type;
+        var widgetModels = [];
+        _.each(data.widgets, function(widgetData) {
 
-        var v = self.addWidget(
-          d.type,
-          _.extend(
-            opts,
-            {
-              datasource: self.datasource
-            }
-          )
-        );
+          var widgetModel = new cdb.geo.ui.Widget.ListModel(widgetData);
```

Comment:
Done [here](https://github.com/CartoDB/cartodb.js/commit/ecde1c4a3b498ca7a5b87605539c56c2c0f9539f). We can move that logic somewhere else.